src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -220,19 +220,22 @@ public async Task GetContentAsync_ErrorStatusCode_ExpectedExceptionThrown(bool w
         }
 
         [Fact]
-        [OuterLoop("Failing connection attempts take long on windows")]
         [SkipOnPlatform(TestPlatforms.Browser, "Socket is not supported on Browser")]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/66692", TestPlatforms.OSX)]
         public async Task GetContentAsync_WhenCanNotConnect_ExceptionContainsHostInfo()
         {
-            using Socket portReserver = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
-            portReserver.Bind(new IPEndPoint(IPAddress.Loopback, 0));
-            IPEndPoint ep = (IPEndPoint)portReserver.LocalEndPoint;
+            const string host = "localhost:1234";
 
-            using var client = CreateHttpClient();
+            using HttpClientHandler handler = CreateHttpClientHandler();
+            handler.ServerCertificateCustomValidationCallback = TestHelper.AllowAllCertificates;
+            var socketsHandler = (SocketsHttpHandler)GetUnderlyingSocketsHttpHandler(handler);
+            socketsHandler.ConnectCallback = (context, token) =>
+            {
+                throw new InvalidOperationException(nameof(GetContentAsync_WhenCanNotConnect_ExceptionContainsHostInfo));
+            };
```

Comment:
This is testing something different now, i.e. an explicitly thrown exception at the beginning of initiating a connection vs an exception coming from the socket layer at some point while failing to connect.  Is that ok?

Reply:
Let me know if you want me to turn it into Theory with sync & async branch @stephentoub. It was not before but it would not be too difficult to do IMHO. Since both cases are handled by same block as @antonfirsov mentioned, behavior should be the same - unless we change the implementation radically.

Reply:
It's up to you.  I just wanted to make sure we were aware of the change to what's being validated.